pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -498,25 +646,35 @@ func (r *hibernationReconciler) setHibernatingCondition(cd *hivev1.ClusterDeploy
 		controllerutils.ErrorScrub(errors.New(message)),
 		controllerutils.UpdateConditionIfReasonOrMessageChange,
 	)
-
-	if reason == hivev1.SyncSetsNotAppliedReason {
-		defer func() {
-			requeueAfter := timeBeforeClusterSyncCheck(cd)
-			logger.Infof("cluster will reconcile due to syncsets not applied in: %v", requeueAfter)
-			result.RequeueAfter = requeueAfter
-			result.Requeue = true
-			returnErr = errors.New(message)
-		}()
+	if changed {
+		logger.WithField("reason", reason).Info("Hibernating condition updated on cluster deployment")
 	}
+	return changed
+}
 
+func (r *hibernationReconciler) setReadyCondition(cd *hivev1.ClusterDeployment, reason, message string, status corev1.ConditionStatus, logger log.FieldLogger) bool {
+	changed := false
+	cd.Status.Conditions, changed = controllerutils.SetClusterDeploymentConditionWithChangeCheck(
+		cd.Status.Conditions,
+		hivev1.ClusterReadyCondition,
+		status,
+		reason,
+		controllerutils.ErrorScrub(errors.New(message)),
+		controllerutils.UpdateConditionIfReasonOrMessageChange,
+	)
 	if changed {
-		if err := r.Status().Update(context.TODO(), cd); err != nil {
-			logger.WithError(err).Log(controllerutils.LogLevel(err), "Failed to update hibernating condition")
-			return reconcile.Result{}, errors.Wrap(err, "failed to update hibernating condition")
-		}
-		logger.WithField("reason", reason).Info("Hibernating condition updated on cluster deployment.")
+		logger.WithField("reason", reason).Info("Ready condition updated on cluster deployment")
 	}
-	return reconcile.Result{}, nil
+	return changed
+}
+
+func (r *hibernationReconciler) updateClusterDeploymentStatus(cd *hivev1.ClusterDeployment, logger log.FieldLogger) error {
+	err := r.Status().Update(context.TODO(), cd)
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to update clusterdeployment")
```

Comment:
(Latent) This use of `LogLevel` is weird. If we failed to update the status, I would think that's worthy of `.Error()` every time.